### PR TITLE
e2e: adapt performance profile existence and stabilize test

### PR DESCRIPTION
**Pull request by rshemtov13**
This PR follows #962 for adding performance profile dependencies and it adds detection for kubeletconfig changes through performance profile. Also this PR adds test id to the test 75354.
This is a reboot test that is changing the kubeletconfig configuration. The changes here are for detecting if a performance profile exists in the first place and changing the configuration through it otherwise the test will fail.

### CODE REVIEW

**Comment by rshemtov13** [comment]
currently wip waiting for #962 to get merged for deps

**Comment by rshemtov13** [comment]
@ffromani

**Comment by ffromani** [comment]
In addition, is the commit message still accurately describing the code changes? I don't see obvious connections now

File: test/e2e/serial/tests/configuration.go [hunk 1]
```diff
@@ -789,10 +789,10 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 			}).WithContext(ctx).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Equal(kcCmNamesPre))
 		})
 
-		It("[reboot_required][slow][unsched][schedrst] should be able to correctly identify topology manager policy without scheduler restarting", Label("reboot_required", "slow", "unsched", "schedrst"), Label("feature:schedattrwatch"), func(ctx context.Context) {
+		It("[test_id:75354][reboot_required][slow][unsched][schedrst] should be able to correctly identify topology manager policy without scheduler restarting", Label("reboot_required", "slow", "unsched", "schedrst"), Label("feature:schedattrwatch"), func(ctx context.Context) {
 			// https://issues.redhat.com/browse/OCPBUGS-34583
 			fxt.IsRebootTest = true
-			By("getting the number of cpus that is required for a numa zone to create a topology affinity error deployment")
+			By("getting the number of cpus that is required for a numa zone to create a Topology Affinity Error deployment")
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:795]: No problem will do from now on

File: test/e2e/serial/tests/configuration.go [hunk 2]
```diff
@@ -820,11 +820,15 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 			klog.Infof("CPU resources requested to create a topology affinity error deployment %s", cpuResources)
 
 			By("fetching the matching kubeletconfig for the numaresources-operator cr")
-			var nroOperObj nropv1.NUMAResourcesOperator
+			nroOperObj := &nropv1.NUMAResourcesOperator{}
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:823]: Yes considering line 828, its better to keep it as is.

File: test/e2e/serial/tests/configuration.go [hunk 3]
```diff
@@ -845,69 +849,152 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
...
+							[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+						))).ToNot(HaveOccurred())
+
+					waitForMcpUpdate(fxt.Client, ctx, mcpsInfo, MachineConfig)
+				}
 			}
 
 			defer func() {
-				err := fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:938]: You're right my mistake. will revert that.

File: test/e2e/serial/tests/configuration.go [hunk 4]
```diff
@@ -845,69 +849,152 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
...
+			err = fxt.Client.List(ctx, profiles)
+			Expect(err).ToNot(HaveOccurred(), "unable to query performance profile")
 
-				Eventually(func(g Gomega) {
-					err := fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+			for _, profile := range profiles.Items {
+				if profile.Kind == "PerformanceProfile" {
+					performanceProfile = profile
+				}
+			}
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:867]: I iterated over the profiles to get the first one. But I can see that I can even delete all that because everywhere in the code where I am changing the TM policy I'm querying it right there so this can actually be removed entirely.

File: test/e2e/serial/tests/configuration.go [hunk 5]
```diff
@@ -74,10 +74,10 @@ import (
...
 
 /*
-MCP gets updated=false & updating=true in 2 cases:
- 1. mc is being updated with new configuration -> new CONFIG: all nodes associated to the pool will be updated
- 2. machine count is increasing: meaning new nodes joins to the pool by matching their roles to the machineConfigSelector.matchExpressions of the mcp.
-    so the same CONFIG on the pool but different machine count. This will trigger reboot on the new nodes thus mcp Updating contidtion will be true.
+	 MCP gets updated=false & updating=true in 2 cases:
+	  1. mc is being updated with new configuration -> new CONFIG: all nodes associated to the pool will be updated
+	  2. machine count is increasing: meaning new nodes joins to the pool by matching their roles to the machineConfigSelector.matchExpressions of the mcp.
+		 so the same CONFIG on the pool but different machine count. This will trigger reboot on the new nodes thus mcp Updating contidtion will be true.
```
Comment (ffromani) [inline test/e2e/serial/tests/configuration.go:80]: this seems unnecessary

**Review by ffromani — COMMENTED** [review]
/approve

code changes looks goos, but I'm still unconvinced the commit message is describing the changes done here. Perhaps needs update?

File: test/e2e/serial/tests/configuration.go (hunk 5 above)
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:80]: Totally unnecessary, probably happened from saving the file for some reason.

File: test/e2e/serial/tests/configuration.go [hunk 6]
```diff
@@ -886,28 +933,59 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 				Expect(err).ToNot(HaveOccurred())
 
 				if initialTopologyManagerPolicy != kcObj.TopologyManagerPolicy {
-					By("reverting kubeletconfig changes")
-					mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
-					Expect(err).ToNot(HaveOccurred())
-					Expect(mcpsInfo).ToNot(BeEmpty())
+					if numTargetedKCOwnerReferences == 0 {
+						By("reverting kuebeletconfig changes to the initial state under kubeletconfig")
+						mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
```
Comment (shajmakh) [inline test/e2e/serial/tests/configuration.go:938]: let's be consistent and use same check flow in all the test, above in https://github.com/openshift-kni/numaresources-operator/pull/977/files#diff-99dd14d7f4fade18212bf91dbe6165d7626335bf407a28732eb7d591d2252fd8R900 the test checks the owner reference then it compares the initial vs current. here it compares the policies and then checks the owner reference let's decide on one way and use that. for example, we use the later in 47585 https://github.com/openshift-kni/numaresources-operator/pull/977/files#diff-99dd14d7f4fade18212bf91dbe6165d7626335bf407a28732eb7d591d2252fd8R900 and then the test calls waitForMcpUpdate once

File: test/e2e/serial/tests/configuration.go [hunk 7]
```diff
@@ -935,51 +1013,89 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
...
 			Expect(err).ToNot(HaveOccurred())
-			Expect(len(pods)).To(Equal(1))
+
+			podsPending := 0
+			for _, pod := range pods {
+				if pod.Status.Phase == corev1.PodPending {
+					podsPending++
+				}
+			}
+			Expect(podsPending).To(Equal(1))
```
Comment (shajmakh) [inline test/e2e/serial/tests/configuration.go:1098]: this should be in a different commit because it solves a different issue than the performance-profile adaptation

Comment (shajmakh) [reply test/e2e/serial/tests/configuration.go:1098]: in that commit scope please add an assertion on the pods' length it is not greater than 10 (so < 10)

**Review by shajmakh — COMMENTED** [review]
Thanks for this!
few comments below

File: test/e2e/serial/tests/configuration.go [hunk 8]
```diff
@@ -789,10 +789,10 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
 			}).WithContext(ctx).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Equal(kcCmNamesPre))
 		})
 
-		It("[reboot_required][slow][unsched][schedrst] should be able to correctly identify topology manager policy without scheduler restarting", Label("reboot_required", "slow", "unsched", "schedrst"), Label("feature:schedattrwatch"), func(ctx context.Context) {
+		It("[test_id:75354][reboot_required][slow][unsched][schedrst] should be able to correctly identify topology manager policy without scheduler restarting", Label("reboot_required", "slow", "unsched", "schedrst"), Label("feature:schedattrwatch"), func(ctx context.Context) {
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:792]: np will add thanks for that!

File: test/e2e/serial/tests/configuration.go [hunk 9]
```diff
@@ -845,37 +849,80 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
...
+				ref := targetedKC.OwnerReferences[0]
+				if ref.Kind != "PerformanceProfile" {
+					e2efixture.Skipf(fxt, "owner object %q is not supported in this test", ref.Kind)
+				}
+
+				if initialTopologyManagerPolicy != v1beta1.NoneTopologyManagerPolicy {
+					By("modifying Topology Manager Policy to 'none' under performance profile")
+					mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
 					Expect(err).ToNot(HaveOccurred())
+					Expect(mcpsInfo).ToNot(BeEmpty())
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:904]: You're right thanks! Will remove the duplicated ones

File: test/e2e/serial/tests/configuration.go [hunk 10]
```diff
@@ -845,37 +849,80 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
...
+					Expect(err).ToNot(HaveOccurred())
 
-				waitForMcpUpdate(fxt.Client, ctx, mcpsInfo, MachineConfig)
+					Expect(fxt.Client.Patch(ctx, updatedProfile,
+						client.RawPatch(
+							types.JSONPatchType,
+							[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+						))).ToNot(HaveOccurred())
+
+					waitForMcpUpdate(fxt.Client, ctx, mcpsInfo, MachineConfig)
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:924]: np will add thanks for that

File: test/e2e/serial/tests/configuration.go (hunk 6 above)
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:938]: You're right it's mixed will choose one consistent way

**Comment by ffromani** [comment]
Looks good, deferring the final LGTM to @shajmakh

File: test/e2e/serial/tests/configuration.go (hunk 7 above)
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:1098]: absolutely right will separate that into a different commit.

File: test/e2e/serial/tests/configuration.go [hunk 11]
```diff
@@ -845,37 +849,80 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
...
-				Eventually(func(g Gomega) {
-					err := fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+			var performanceProfile perfprof.PerformanceProfile
+
+			if numTargetedKCOwnerReferences == 0 {
+				if initialTopologyManagerPolicy != v1beta1.NoneTopologyManagerPolicy {
+					By("modifying Topology Manager Policy to 'none' under kubeletconfig")
+					mcpsInfo, err := buildMCPsInfo(fxt.Client, ctx, nroOperObj)
 					Expect(err).ToNot(HaveOccurred())
+					Expect(mcpsInfo).ToNot(BeEmpty())
```
[the remark this answers is not in the document]
Comment (rshemtov13) [reply test/e2e/serial/tests/configuration.go:874]: Thanks for that. Changed it

**Comment by rshemtov13** [comment]
Closing this PR because rebasing was done incorrectly and It would take to long to resolve the conflicts here.
Will reference the new PR for the changes here because the review for the changes was made here.

**Comment by rshemtov13** [comment]
New PR for these changes are in #981
